umap/static/umap/js/umap.controls.js

```diff
@@ -1252,16 +1252,40 @@ L.U.Editable = L.Editable.extend({
         var content;
         if (e.layer instanceof L.Marker) content = L._('Click to add a marker');
         else if (e.layer instanceof L.Polyline) {
-            if (!e.layer.editor._drawnLatLngs.length) {
-                if (e.layer instanceof L.Polygon) content = L._('Click to start drawing a polygon');
-                else if (e.layer instanceof L.Polyline) content = L._('Click to start drawing a line');
-            } else if (e.layer.editor._drawnLatLngs.length < e.layer.editor.MIN_VERTEX) {
-                content = L._('Click to continue drawing');
+            // when drawing a Polyline or Polygon
+            if (e.layer.editor._drawnLatLngs) {
+                // when drawing first point
+                if (!e.layer.editor._drawnLatLngs.length) {
+                    if (e.layer instanceof L.Polygon){
+                        content = L._('Click to start drawing a polygon');
+                    } else if (e.layer instanceof L.Polyline) {
+                        content = L._('Click to start drawing a line');
+                    }
+                } else {
+                    var tmpLatLngs = e.layer.editor._drawnLatLngs.slice();
+                    tmpLatLngs.push(e.latlng);
+                    var readableDistance = e.layer.getMeasure(tmpLatLngs);
+                    if (e.layer.editor._drawnLatLngs.length < e.layer.editor.MIN_VERTEX) {
+                        // when drawing second point
+                        content = L._('Click to continue drawing');
+                    }  else {
+                        // when drawing third point (or more)
+                        content = L._('Click last point to finish shape');
+                    }
+                    content += " ("+readableDistance+")";
+                }
             } else {
-                content = L._('Click last point to finish shape');
+                // when moving an existing point
+                if (e.layer instanceof L.Polygon){
+                    content = L._('Polygon area: {measure}', { measure: e.layer.getMeasure() });
+                } else if (e.layer instanceof L.Polyline) {
+                    content = L._('Line distance: {measure}', { measure: e.layer.getMeasure() });
```

Comment:
No sure about my English, but I feal like this should be `Line length` instead of `Line distance`, what do you think ?

Also, at this stage, I think you can leave out the translations, this will be done automatically by Transifex later in the process.